plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestStandardColumnMappings.java

```diff
@@ -0,0 +1,127 @@
+        picosOfMicro += (instant.getNano() % NANOSECONDS_PER_MICROSECOND) * PICOSECONDS_PER_NANOSECOND;
+        verifyShortTimestamp(epochMicros, picosOfMicro, precision);
+    }
+
+    @DataProvider
+    public Object[][] validShortTimestamps()
+    {
+        return new Object[][] {
+                {0, LocalDateTime.of(2018, 3, 25, 3, 17, 17), 0},
+                {1, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 100_000_000), 0},
+                {2, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 120_000_000), 0},
+                {3, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_000_000), 0},
+                {4, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_400_000), 0},
+                {5, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_450_000), 0},
+                {6, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_456_000), 0},
+        };
+    }
+
+    @Test(dataProvider = "validLongTimestamps")
+    public void testVerifyValidLongTimestamps(int precision, LocalDateTime localDateTime, int picosOfMicro)
+    {
+        Instant instant = localDateTime.atZone(UTC).toInstant();
+        long epochMicros = instant.getEpochSecond() * MICROSECONDS_PER_SECOND + floorDiv(instant.getNano(), NANOSECONDS_PER_MICROSECOND);
+        picosOfMicro += (instant.getNano() % NANOSECONDS_PER_MICROSECOND) * PICOSECONDS_PER_NANOSECOND;
+        verifyLongTimestamp(picosOfMicro, precision);
+    }
+
+    @DataProvider
+    public Object[][] validLongTimestamps()
+    {
+        return new Object[][] {
+                {7, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_456_700), 0},
+                {8, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_456_780), 0},
+                {9, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_456_789), 0},
+                {10, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_456_789), 100},
+                {11, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_456_789), 120},
+                {12, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 123_456_789), 123},
+        };
+    }
+
+    @Test(dataProvider = "invalidShortTimestamps")
+    public void testVerifyInvalidShortTimestamps(int precision, LocalDateTime localDateTime, int picosOfMicro, Class<?> expectedException)
+    {
+        Instant instant = localDateTime.atZone(UTC).toInstant();
+        long epochMicros = instant.getEpochSecond() * MICROSECONDS_PER_SECOND + floorDiv(instant.getNano(), NANOSECONDS_PER_MICROSECOND);
+        int finalPicosOfMicro = picosOfMicro + (instant.getNano() % NANOSECONDS_PER_MICROSECOND) * PICOSECONDS_PER_NANOSECOND;
+        assertThatThrownBy(() -> verifyShortTimestamp(epochMicros, finalPicosOfMicro, precision)).isInstanceOf(expectedException);
+    }
+
+    @DataProvider
+    public Object[][] invalidShortTimestamps()
+    {
+        return new Object[][] {
+                {0, LocalDateTime.of(2018, 3, 25, 3, 17, 17, 100_000_000), 0, VerifyException.class},
```

Comment:
nit: I would prefer to have meaningful part of exception message instead of exception class here. And use `hasMessageContaining` in the test method.

Reply:
Looks _much_ better, thanks!